lib/data/wrapper.js

```diff
@@ -153,11 +153,19 @@ const data = {
 
     // It would be preferable to have an sproxyd batch delete route to
     // replace this
-    batchDelete: (locations, log) => {
+    batchDelete: (locations, requestMethod, newObjDataStoreName, log) => {
         // TODO: The method of persistence of sproxy delete key will
         // be finalized; refer Issue #312 for the discussion. In the
         // meantime, we at least log the location of the data we are
         // about to delete before attempting its deletion.
+        const skipBackend = ['aws_s3'];
+        const isSkipBackend = (locations[0] && locations[0].dataStoreType) ?
+            skipBackend.indexOf(locations[0].dataStoreType) > -1 : false;
+        const isMatchingBackends = locations[0] ?
+            locations[0].dataStoreName === newObjDataStoreName : false;
+        if (requestMethod === 'PUT' && isSkipBackend && isMatchingBackends) {
```

Comment:
Why is this new check necessary? It does not come straightforward, maybe a comment would be useful. Can it make more sense to have the check supported by a backend-specific batchDelete?